tensorflow_datasets/core/load.py

```diff
@@ -179,6 +180,65 @@ def builder(
     return builder_cls(name)(**builder_kwargs)  # pytype: disable=not-instantiable
 
 
+def find_builder(name: str, *, data_dir: Optional[str] = None) -> Optional[str]:
+  """Returns the path to dataset files
+
+  Example:
+
+  ```py
+  path = find_builder_dir('ds/config:1.2.0', data_dir='~/tensorflow_datasets/')
+  asssert path == '~/tensorflow_datasets/ds/config/1.2.0'
+
+  # If the version is not specified, return the last available version
+  find_builder_dir('ds', data_dir='~/tensorflow_datasets/')
+  assert path == '~/tensorflow_datasets/my_dataset/config/1.2.0'
+  ```
+
+  """
+  ds_name, name_kwargs = _dataset_name_and_kwargs_from_name_str(name)
+
+  base_dir = os.path.join(data_dir or constants.DATA_DIR, ds_name)
+  config = name_kwargs.get("config", None)
+  version = name_kwargs.get("version", None)
+
+  if config and version:
+    builder_dir = os.path.join(base_dir, config, version)
+    if tf.io.gfile.exists(builder_dir):
+      return builder_dir
+
+  if version:
+    builder_dir = os.path.join(base_dir, version)
+    if tf.io.gfile.exists(builder_dir):
+      return builder_dir
+
+  config_version = _select_config_and_version(base_dir)
+  if config_version:
+    return os.path.join(base_dir, config_version)
+
+  version = _select_version(base_dir)
+  if version:
+    return os.path.join(base_dir, version)
+
+  return None
+
+def _select_config_and_version(data_dir: str) -> Optional[str]:
+  config = tf.io.gfile.listdir(data_dir)[0]  # TODO: Select default config
+  if config:
+    version = _select_version(os.path.join(data_dir, config))
+    if version:
+      return posixpath.join(config, version)
+  return None
+
+
+def _select_version(data_dir: str) -> Optional[str]:
+  version_regex = re.compile(r"^[0-9]+\.[0-9]+\.[0-9]+$")
+  dirs = sorted(
```

Comment:
This will fail: `['9.0.0', '10.0.0']` will return `9.0.0` instead of `10.0.0`. Could you add a test for this ?

You can use `Version('1.0.0')` as `>` will work correctly between two version objects

Also look at the DatasetBuilder version code to see how this is done and whether you can reuse part of the code.